controllers/openstackdataplanenodeset_controller.go

```diff
@@ -257,39 +241,45 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 	// Reconcile BaremetalSet if required
 	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) > 0 {
 		isReady, err := deployment.DeployBaremetalSet(ctx, helper, instance,
-			nodes, allIPSets, dnsAddresses)
+			allIPSets, dnsAddresses)
 		if err != nil || !isReady {
 			return ctrl.Result{}, err
 		}
 	}
 
-	// Ensure all nodes are in SetupReady state
-	setupReadyNodes := 0
-	for _, node := range nodes.Items {
-		if node.IsSetupReady() {
-			setupReadyNodes++
-		}
-	}
-
-	if setupReadyNodes < len(nodes.Items) {
-		return ctrl.Result{}, err
-	}
-
 	// Generate Role Inventory
 	roleConfigMap, err := deployment.GenerateRoleInventory(ctx, helper, instance,
-		nodes.Items, allIPSets, dnsAddresses)
+		allIPSets, dnsAddresses)
 	if err != nil {
 		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to generate inventory for %s", instance.Name), instance)
 		return ctrl.Result{}, err
 	}
 
+	// Verify Ansible SSH Secret
+	ansibleSSHPrivateKeySecret = instance.Spec.AnsibleSSHPrivateKeySecret
+	if ansibleSSHPrivateKeySecret != "" {
+		_, result, err = secret.VerifySecret(
+			ctx,
+			types.NamespacedName{Namespace: instance.Namespace, Name: ansibleSSHPrivateKeySecret},
+			[]string{
+				"ssh-privatekey",
+			},
+			r.Client,
+			time.Duration(5)*time.Second,
+		)
+		if err != nil {
+			return result, err
+		}
+	}
+
 	// all setup tasks complete, mark SetupReadyCondition True
 	instance.Status.Conditions.MarkTrue(dataplanev1.SetupReadyCondition, condition.ReadyMessage)
 
-	r.Log.Info("Role", "DeployStrategy", instance.Spec.DeployStrategy.Deploy,
-		"Role.Namespace", instance.Namespace, "Role.Name", instance.Name)
-	if instance.Spec.DeployStrategy.Deploy {
-		r.Log.Info("Starting DataPlaneRole deploy")
+	// Trigger executions based on the OpenStackDataPlane Controller state.
+	_, readyToExecute := instance.ObjectMeta.Labels["OpenStackDataPlaneManagedBy"]
+	if readyToExecute {
+		r.Log.Info("Deploying NodeSet: %s", instance.Name)
+		r.Log.Info("Starting DataPlaneNodeSet deploy")
```

Comment:
We need to discuss how we're going to handle triggering a deployment. Let's leave DeployStrategy as-is for now, so we can make progress on the simplification outside of that change.

Reply:
Ah, this answers the question I asked in the last comment.

I'll revert it for now and we'll have it in both controllers.